---
title: "Call Variants with SAMtools Element"
weight: 100
---


# Call Variants with SAMtools Element

Calls SNPs and INDELS with SAMtools mpileup and bcftools.

**Element type:** call\_variants

Parameters
----------

Parameter

Description

Default value

Parameter in Workflow File

Type

**Output variants file**

The url to the file with the extracted variations.







**Reference**

Specify a file with the reference sequence.

The sequence will be used as reference for all datasets with NGS assemblies.







**Use reference from**

Specify "File" to set a single reference sequence for all input NGS assemblies. The reference should be set in the "Reference" parameter.

Specify "Input port" to be able to set different references for difference NGS assemblies. The references should be input via the "Input sequences" port (e.g. use datasets in the "Read Sequence" element).

File





**Illumina-1.3+ encoding**

Assume the quality is in the Illumina 1.3+ encoding (mpileup)(-6).

False

**illumina13-encoding**

_boolean_

**Count anomalous read pairs**

Do not skip anomalous read pairs in variant calling (mpileup)(-A).

 False

**use\_orphan**

_boolean_

**Disable BAQ computation**

Disable probabilistic realignment for the computation of base alignment quality (BAQ). BAQ is the Phred-scaled probability of a read base being misaligned. Applying this option greatly helps to reduce false SNPs caused by misalignments (mpileup)(-B).

 False

**disable\_baq**

_boolean_

**Mapping quality downgrading coefficient**

Coefficient for downgrading mapping quality for reads containing excessive mismatches. Given a read with a phred-scaled probability q of being generated from the mapped position, the new mapping quality is about sqrt((INT-q)/INT)\*INT. A zero value disables this functionality; if enabled, the recommended value for BWA is 50 (mpileup)(-C).

0

**capq\_thres**

_numeric_

**Max number of reads per input BAM**

At a position, read maximally the number of reads per input BAM (mpileup)(-d).

250

**max\_depth**

_numeric_

**Extended BAQ computation**

Extended BAQ computation. This option helps sensitivity especially for MNPs, but may hurt specificity a little bit (mpileup)(-E).

False

**ext\_baq**

_boolean_

**BED or position list file**

BED or position list file containing a list of regions or sites where pileup or BCF should be generated. (mpileup)(-l).



**bed**

_string_

**Pileup region**

Only generate pileup in region STR (mpileup)(-r).



**reg**

_string_

**Minimum mapping quality**

Minimum mapping quality for an alignment to be used (mpileup)(-q).

0

**min\_mq**

_numeric_

**Minimum base quality**

Minimum base quality for a base to be considered (mpileup)(-Q).

13

**min\_baseq**

_numeric_

**Gap extension error**

Phred-scaled gap extension sequencing error probability. Reducing INT leads to longer indels (mpileup)(-e).

20

**extQ**

_numeric_

**Homopolymer errors coefficient**

Coefficient for modeling homopolymer errors. Given an l-long homopolymer run, the sequencing error of an indel of size s is modeled as INT\*s/l. (mpileup)(-h).

100

**tandemQ**

_numeric_

**No INDELs**

Do not perform INDEL calling (mpileup)(-I).

False

**no\_indel**

_boolean_

**Max INDEL depth**

Skip INDEL calling if the average per-sample depth is above INT (mpileup)(-L).

250

**max\_indel\_depth**

_numeric_

**Gap open error**

Phred-scaled gap open sequencing error probability. Reducing INT leads to more indel calls (mpileup)(-o).

40

**openQ**

_numeric_

**List of platforms for indels**

Comma dilimited list of platforms (determined by @RG-PL) from which indel candidates are obtained.It is recommended to collect indel candidates from sequencing technologies that have low indel error rate such as ILLUMINA. (mpileup)(-P).



**pl\_list**

_string_

**Retain all possible alternate**

Retain all possible alternate alleles at variant sites. By default, the view command discards unlikely alleles. (bcf view)(-A).

False

**keepalt**

_boolean_

**Indicate PL**

Indicate PL is generated by r921 or before (ordering is different) (bcf view)(-F).

False

**fix\_pl**

_boolean_

**No genotype information**

Suppress all individual genotype information (bcf view)(-G).

False

**no\_geno**

_boolean_

**A/C/G/T only**

Skip sites where the REF field is not A/C/G/T (bcf view)(-N).

False

**acgt\_only**

_boolean_

**List of sites**

List of sites at which information are outputted (bcf view)(-l).



**bcf\_bed**

_string_

**QCALL likelihood**

Output the QCALL likelihood format (bcf view)(-Q).

False

**qcall**

_boolean_

**List of samples**

List of samples to use. The first column in the input gives the sample names and the second gives the ploidy, which can only be 1 or 2. When the 2nd column is absent, the sample ploidy is assumed to be 2. In the output, the ordering of samples will be identical to the one in FILE (bcf view)(-s).



**samples**

_string_

**Min samples fraction**

skip loci where the fraction of samples covered by reads is below FLOAT (bcf view)(-d).

0

**min\_smpl\_frac**

_numeric_

**Per-sample genotypes**

Call per-sample genotypes at variant sites. (bcf view)(-g).

True

**call\_gt**

_boolean_

**INDEL-to-SNP Ratio**

Ratio of INDEL-to-SNP mutation rate. (bcf view)(-i).

\-1

**indel\_frac**

_numeric_

**Max P(ref|D)**

A site is considered to be a variant if P(ref|D)

0.5

**pref**

_numeric_

**Prior allele frequency spectrum**

If STR can be full, cond2, flat or the file consisting of error output from a previous variant calling run (bcf view)(-P).

full

**ptype**

_string_

**Mutation rate**

Scaled mutation rate for variant calling (bcf view)(-t).

0.001

**theta**

_numeric_

**Pair/trio calling**

Enable pair/trio calling. For trio calling, option -s is usually needed to be applied to configure the trio members and their ordering. In the file supplied to the option -s, the first sample must be the child, the second the father and the third the mother. The valid values of STR are pair, trioauto, trioxd and trioxs, where pair calls differences between two input samples, and trioxd (trioxs)specifies that the input is from the X chromosome non-PAR regions and the child is a female (male) (bcf view)(-T).



**ccall**

_string_

**N group-1 samples**

Number of group-1 samples. This option is used for dividing the samples into two groups for contrast SNP calling or association test. When this option is in use, the followingVCF INFO will be outputted: PC2, PCHI2 and QCHI2 (bcf view)(-1).

0

**n1**

_numeric_

**N permutations**

Number of permutations for association test (effective only with -1) (bcf view)(-U).

0

**n\_perm**

_numeric_

**Min P(chi^2)**

Only perform permutations for P(chi^2).

0.01

**min\_perm\_p**

_numeric_

**Minimum RMS quality**

Minimum RMS mapping quality for SNPs (varFilter) (-Q).

10

**min-qual**

_numeric_

**Minimum read depth**

Minimum read depth (varFilter) (-d).

2

**min-dep**

_numeric_

**Maximum read depth**

Maximum read depth (varFilter) (-D).

10000000

**max-dep**

_numeric_

**Alternate bases**

Minimum number of alternate bases (varFilter) (-a).

2

**min-alt-bases**

_numeric_

**Gap size**

SNP within INT bp around a gap to be filtered (varFilter) (-w).

3

**gap-size**

_numeric_

**Window size**

Window size for filtering adjacent gaps (varFilter) (-W).

10

**window"**

_numeric_

**Strand bias**

Minimum P-value for strand bias (given PV4) (varFilter) (-1).

0.0001

**min-strand**

_numeric_

**BaseQ bias**

Minimum P-value for baseQ bias (varFilter) (-2).

1e-100

**min-baseQ**

_string_

**MapQ bias**

Minimum P-value for mapQ bias (varFilter) (-3).

0

**min-mapQ**

_numeric_

**End distance bias**

Minimum P-value for end distance bias (varFilter) (-4).

0.0001

**min-end-distance**

_numeric_

**HWE**

Minimum P-value for HWE (plus F).

0.0001

**min-hwe**

_numeric_

**Log filtered**

Print filtered variants into the log (varFilter) (-p).

False

**print-filtered**

_boolean_

Input/Output Ports
------------------

The element has 2 _input ports_:

**Name in GUI:** Input assembly

**Name in Workflow File:** in-assembly

**Slots:**

Slot In GUI

Slot in Workflow File

Type

**Dataset name**

**dataset**

_string_

**Source url**

**url**

_string_

**Name in GUI:** Input sequences

**Name in Workflow File:** in-sequence

**Slots:**

Slot In GUI

Slot in Workflow File

Type

**Source url**

**url**

_string_

And 1 _output port_:

**Name in GUI:** Output variations

**Name in Workflow File:** out-variations

**Slots:**

Slot In GUI

Slot in Workflow File

Type

**Variation track**

**variation-track**

_variation_
